### PR TITLE
[GUILD] TypeScript `EmailSelector`

### PR DESCRIPTION
**Pull request by lowiebenoot**
This PR converts the `EmailSelector` component to typescript.
It was quite some work to find out all the correct types, so I could definitely use some good eyes on this.

### CODE REVIEW

File: src/components/emailSelector/Autocomplete.tsx [hunk 1]
```diff
@@ -1,14 +1,28 @@
...
 import { Heading4 } from '../typography';
 import EmailSuggestion from './EmailSuggestion';
+import { Suggestion } from './types';
 
-const Autocomplete = ({ suggestions, highlightedSuggestion, onClick, onHover, renderSuggestion }) => {
+interface AutocompleteProps {
+  suggestions: Suggestion[] | { [x: string]: Suggestion[] };
+  highlightedSuggestion?: Suggestion | null;
+  onClick?: (suggestion: Suggestion, event: React.SyntheticEvent) => void;
+  onHover?: (suggestion: Suggestion, event: React.SyntheticEvent) => void;
```
Comment (BeirlaenAaron) [inline src/components/emailSelector/Autocomplete.tsx:15]: These should be required

**Review by BeirlaenAaron — DISMISSED** [review]

File: src/components/emailSelector/Autocomplete.tsx (hunk 1 above)
Comment (lowiebenoot) [reply src/components/emailSelector/Autocomplete.tsx:15]: They are used as `onMouseOver={onHover?.bind(null, suggestion)}` so they seem optional to me?

**Comment by lorgan3** [comment]
Haven't reviewed yet but you could reference [this](https://github.com/teamleadercrm/deals-frontend/tree/email-selector-do-not-delete-until-tests-are-moved-to-ui-lib/src/Lib/Components/EmailSelector) for the types since that what the implementation here is based on 😅

**Review by lorgan3 — DISMISSED** [review]

**Review by lorgan3 — DISMISSED** [review]

**Review by ArnaudWeyts — DISMISSED** [review]

**Review by qubis741 — APPROVED** [review]

**Comment by qubis741** [comment]
Manually checked :white_check_mark:

**Review by lorgan3 — APPROVED** [review]

**Pull request merged**